ethcore/src/executive.rs

```diff
@@ -291,7 +291,7 @@ impl<'a, B: 'a + StateBackend, E: Engine + ?Sized> Executive<'a, B, E> {
 		let mut substate = Substate::new();
 
 		// NOTE: there can be no invalid transactions from this point.
-		if !t.is_unsigned() {
+		if !schedule.eip86 || !t.is_unsigned() {
```

Comment:
clarify: if EIP86 is not enabled then no unsigned transactions can reach this point?

Reply:
if EIP86 is not enabled then no unsigned transactions can reach this because this would require signing a transaction from zero address. Still there's a test that simulates such transaction under pre-constantinople rules, hence the additional check. As for the balance decrease - unsigned transaction must have value == 0